### PR TITLE
🔥 Remove `tipsy` tooltips

### PR DESCRIPTION
**Pull request by josemigallas**
`tipsy` is a jQuery plugin that shows a tooltip to HTML elements with a `title`, .e.g. `<input title="Show me a tooltip" />`

It's currently being used in:
* provider signup, not top be used anymore, and also stupid:
Screenshot 2024-06-05 at 08 29 26

* cms, redundant and also stupid:
Screenshot 2024-06-05 at 08 30 23

* theoretically supported in dev portal, but package `tipsy` is not imported anywhere and I believe it's just this line has been carried over from version to version because nobody bothered to clean it up: https://github.com/3scale/porta/blob/505d6ccfb919d2a20eeb9cf7474468b94a149c60/lib/developer_portal/app/views/developer_portal/javascripts/3scale_v2.js#L757-L762

Moreover, it seems like it's not supported by latest jQuery + jQuery-ui:
Screenshot 2024-06-05 at 09 43 21

IMHO this tipsy thing is outdated, adds tech debt and doesn't add any UX value. Besides, we already have Patternfly Popover implemented for the admin portal, when needed:
Screenshot 2024-06-05 at 09 48 04

And the dev portal has Bootstrap, which also features a tooltip:
Screenshot 2024-06-05 at 09 52 44

In the CMS, the default tooltips look like this:
image

### CODE REVIEW

**Review by jlledom — COMMENTED** [review]
I agree on removing this, as we already have a Patternfly component for it. Also, we don't have to worry about the provider signup screen anymore. Although I think Tipsy should be replaced by the Patternfly tooltip component in the CMS screen.

Also, not sure if this would break any developer portal. If the only reference to tipsy in the dev. portal code is this I don't think removing it will break anything:

```javascript
    // Enable Tipsy for links with title to show nice and fast tooltips with title
    if($.fn.tipsy) {
      // can't enable live, because mouseleave is not called
      $('[title]').tipsy({gravity: $.fn.tipsy.autoWE});
      $('time').tipsy({gravity: $.fn.tipsy.autoWE})
    }
```

**Comment by josemigallas** [comment]
> I agree on removing this, as we already have a Patternfly component for it. Also, we don't have to worry about the provider signup screen anymore. Although I think Tipsy should be replaced by the Patternfly tooltip component in the CMS screen.

IMHO I don't think it's not worth the effort. The browser already features a builtin tooltip, like this:
Screenshot 2024-06-07 at 11 34 40

At least right now I wouldn't do it.

> Also, not sure if this would break any developer portal. If the only reference to tipsy in the dev. portal code is this I don't think removing it will break anything:
> ```js
>     // Enable Tipsy for links with title to show nice and fast tooltips with title
>     if($.fn.tipsy) {
>       // can't enable live, because mouseleave is not called
>       $('[title]').tipsy({gravity: $.fn.tipsy.autoWE});
>       $('time').tipsy({gravity: $.fn.tipsy.autoWE})
>     }
> ```

I was worried that tipsy was provided as an asset in the past, but it seems it was not, therefore we are safe. This small snippet comes from an old script that was used in the admin portal that somehow got liked into then dev portal.

**Review by mayorova — DISMISSED** [review]
LGTM

File: app/assets/javascripts/vendor/jquery.tipsy.js [hunk 1]
```diff
@@ -1,337 +1 @@
...
-      if ($(window).width() + $(document).scrollLeft() - $this.offset().left < marginEast) dir.ew = 'e';
-      if ($(window).height() + $(document).scrollTop() - $this.offset().top < marginNorth) dir.ns = 's';
-
-      return dir.ns + (dir.ew ? dir.ew : '');
-    };
-  };
-
-})(jQuery);
\ No newline at end of file
+// TODO: make sure this is not imported in any dev portal
```
Comment (lvillen) [inline app/assets/javascripts/vendor/jquery.tipsy.js:1]: According to your last reply to Joan, maybe we can even remove this file?

File: app/javascript/packs/cms.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@ import $ from 'jquery'
 import 'jquery-ui/ui/widgets/droppable'
 import 'jquery-ui/ui/widgets/draggable'
 import 'jquery-ui/ui/widgets/tabs'
+import 'jquery-ui/ui/widgets/tooltip'
```
Comment (lvillen) [inline app/javascript/packs/cms.ts:5]: What do we need this `tooltip` for?

**Review by lvillen — COMMENTED** [review]
Had a couple questions but LGTM!

File: app/assets/javascripts/vendor/jquery.tipsy.js (hunk 1 above)
Comment (josemigallas) [reply app/assets/javascripts/vendor/jquery.tipsy.js:1]: Ah yes, good catch!

File: app/javascript/packs/cms.ts (hunk 2 above)
Comment (josemigallas) [reply app/javascript/packs/cms.ts:5]: Yet another leftover from some testing :)

**Pull request merged**